tools/crm_mon.c

```diff
@@ -689,11 +690,13 @@ reconnect_after_timeout(gpointer data)
     }
 
     print_as(output_format, "Reconnecting...\n");
-    fencing_connect();
-    if (cib_connect(TRUE) == pcmk_ok) {
-        /* Redraw the screen and reinstall ourselves to get called after another reconnect_msec. */
-        mon_refresh_display(NULL);
-        return FALSE;
+    if (pacemakerd_status() == pcmk_rc_ok) {
+        fencing_connect();
+        if (cib_connect(TRUE) == pcmk_ok) {
+            /* Redraw the screen and reinstall ourselves to get called after another reconnect_msec. */
+            mon_refresh_display(NULL);
+            return FALSE;
+        }
```

Comment:
You'll probably need to add a call to pacemakerd_status() into the one_shot function and after the refresh label in detect_user_input.  Now that I see how often the same code block is being called, it makes me think I should have put it into a function.

Reply:
Was quite hesitant with those because we should maybe print something different.
But maybe you are right. That just unconditionally calling it before and in any case call the other connect functions afterwards - in any case even when we expect them to fail - will give us the printout we need.
Or I print out that cib and fencer are not available if pacemakerd_status returns ENOTCONN and don't call the others.